vec3.h

```diff
@@ -0,0 +1,120 @@
+//code by Ameya Vikrama Singh
+#include<iostream>
+#include<cmath>
+class vec3{
+    public:
+        float x,y,z;
```

Comment:
A nice enhancement would be to make it so that the user can easily switch between `float` and `double` vectors. Maybe a templated `vec3` class or a preprocessor macro?

Reply:
Would it be better to create a file say `config.cpp` which contains all such macros we may need?